// appium-support/build/lib/fs.js
'use strict';

var _regeneratorRuntime = require('babel-runtime/regenerator')['default'];

var _getIterator = require('babel-runtime/core-js/get-iterator')['default'];

var _interopRequireDefault = require('babel-runtime/helpers/interop-require-default')['default'];

Object.defineProperty(exports, '__esModule', {
  value: true
});

var _fs2 = require('fs');

var _fs3 = _interopRequireDefault(_fs2);

var _rimraf = require('rimraf');

var _rimraf2 = _interopRequireDefault(_rimraf);

var _md52 = require('md5');

var _md53 = _interopRequireDefault(_md52);

var _ncp = require('ncp');

var _ncp2 = _interopRequireDefault(_ncp);

var _bluebird = require('bluebird');

var _bluebird2 = _interopRequireDefault(_bluebird);

var _mv = require('mv');

var _mv2 = _interopRequireDefault(_mv);

var fs = {
  hasAccess: function hasAccess(path) {
    return _regeneratorRuntime.async(function hasAccess$(context$1$0) {
      while (1) switch (context$1$0.prev = context$1$0.next) {
        case 0:
          context$1$0.prev = 0;
          context$1$0.next = 3;
          return _regeneratorRuntime.awrap(this.access(path, fs.F_OK | fs.R_OK));

        case 3:
          context$1$0.next = 8;
          break;

        case 5:
          context$1$0.prev = 5;
          context$1$0.t0 = context$1$0['catch'](0);
          return context$1$0.abrupt('return', false);

        case 8:
          return context$1$0.abrupt('return', true);

        case 9:
        case 'end':
          return context$1$0.stop();
      }
    }, null, this, [[0, 5]]);
  },
  exists: function exists(path) {
    return this.hasAccess(path);
  },
  rimraf: _bluebird2['default'].promisify(_rimraf2['default']),
  mkdir: function mkdir(dirName) {
    var _mkdir;

    return _regeneratorRuntime.async(function mkdir$(context$1$0) {
      while (1) switch (context$1$0.prev = context$1$0.next) {
        case 0:
          _mkdir = _bluebird2['default'].promisify(_fs3['default'].mkdir);
          context$1$0.prev = 1;
          context$1$0.next = 4;
          return _regeneratorRuntime.awrap(_mkdir(dirName));

        case 4:
          context$1$0.next = 10;
          break;

        case 6:
          context$1$0.prev = 6;
          context$1$0.t0 = context$1$0['catch'](1);

          if (!(context$1$0.t0 && context$1$0.t0.code !== "EEXIST")) {
            context$1$0.next = 10;
            break;
          }

          throw context$1$0.t0;

        case 10:
        case 'end':
          return context$1$0.stop();
      }
    }, null, this, [[1, 6]]);
  },
  copyFile: _bluebird2['default'].promisify(_ncp2['default']),
  md5: function md5(filePath) {
    return _regeneratorRuntime.async(function md5$(context$1$0) {
      while (1) switch (context$1$0.prev = context$1$0.next) {
        case 0:
          context$1$0.next = 2;
          return _regeneratorRuntime.awrap(fs.readFile(filePath));

        case 2:
          context$1$0.t0 = context$1$0.sent;
          return context$1$0.abrupt('return', (0, _md53['default'])(context$1$0.t0));

        case 4:
        case 'end':
          return context$1$0.stop();
      }
    }, null, this);
  },
  mv: _bluebird2['default'].promisify(_mv2['default'])
};

var simples = ['open', 'close', 'access', 'readFile', 'writeFile', 'write', 'readlink', 'chmod', 'unlink', 'readdir', 'stat', 'rename', 'lstat'];

var _iteratorNormalCompletion = true;
var _didIteratorError = false;
var _iteratorError = undefined;

try {
  for (var _iterator = _getIterator(simples), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = true) {
    var s = _step.value;

    fs[s] = _bluebird2['default'].promisify(_fs3['default'][s]);
  }
} catch (err) {
  _didIteratorError = true;
  _iteratorError = err;
} finally {
  try {
    if (!_iteratorNormalCompletion && _iterator['return']) {
      _iterator['return']();
    }
  } finally {
    if (_didIteratorError) {
      throw _iteratorError;
    }
  }
}

exports['default'] = fs;
module.exports = exports['default'];
