// appium-support/build/lib/system.js
'use strict';

var _regeneratorRuntime = require('babel-runtime/regenerator')['default'];

var _interopRequireDefault = require('babel-runtime/helpers/interop-require-default')['default'];

Object.defineProperty(exports, '__esModule', {
  value: true
});

var _teen_process = require('teen_process');

var _os = require('os');

var _os2 = _interopRequireDefault(_os);

function isWindows() {
  return _os2['default'].type() === 'Windows_NT';
}

function isMac() {
  return _os2['default'].type() === 'Darwin';
}

function isLinux() {
  return !isWindows() && !isMac();
}

function isOSWin64() {
  return process.arch === 'x64' || process.env.hasOwnProperty('PROCESSOR_ARCHITEW6432');
}

function arch() {
  var _ref, stdout, is64;

  return _regeneratorRuntime.async(function arch$(context$1$0) {
    while (1) switch (context$1$0.prev = context$1$0.next) {
      case 0:
        if (!(isLinux() || isMac())) {
          context$1$0.next = 8;
          break;
        }

        context$1$0.next = 3;
        return _regeneratorRuntime.awrap((0, _teen_process.exec)('uname', ['-m']));

      case 3:
        _ref = context$1$0.sent;
        stdout = _ref.stdout;
        return context$1$0.abrupt('return', stdout.trim() === 'i686' ? '32' : '64');

      case 8:
        if (!isWindows()) {
          context$1$0.next = 11;
          break;
        }

        is64 = this.isOSWin64();
        return context$1$0.abrupt('return', is64 ? '64' : '32');

      case 11:
      case 'end':
        return context$1$0.stop();
    }
  }, null, this);
}

exports.isWindows = isWindows;
exports.isMac = isMac;
exports.isLinux = isLinux;
exports.isOSWin64 = isOSWin64;
exports.arch = arch;
